app/services/preserved_object_handler.rb

```diff
@@ -60,7 +60,66 @@ def create
   end
 
   def check_existence
-    # this is just a stub
+    if invalid?
+      handler_results.add_result(PreservedObjectHandlerResults::INVALID_ARGUMENTS, errors.full_messages)
+    elsif PreservedObject.exists?(druid: druid)
+      Rails.logger.debug "check_existence #{druid} called"
+      if endpoint.endpoint_type.endpoint_class == 'online'
+        transaction_ok = with_active_record_transaction_and_rescue do
+          pres_object = PreservedObject.find_by!(druid: druid)
+          # FIXME: what if there is more than one associated pres_copy?
+          pres_copy = PreservedCopy.find_by!(preserved_object: pres_object, endpoint: endpoint) if pres_object
+
+          if pres_copy.version != pres_object.current_version
+            handler_results.add_result(PreservedObjectHandlerResults::PC_PO_VERSION_MISMATCH, pres_copy.class.name)
+            raise ActiveRecord::Rollback, 'PO current_version != PC version'
+          end
+
+          if incoming_version == pres_copy.version
+            handler_results.add_result(PreservedObjectHandlerResults::VERSION_MATCHES, pres_copy.class.name)
+            handler_results.add_result(PreservedObjectHandlerResults::VERSION_MATCHES, pres_object.class.name)
+            update_pc_validation_timestamps(pres_copy)
+            update_db_object(pres_copy)
```

Comment:
Do we want to update `pres_obj` as well? Right now `pres_obj` is only updated when the incoming version is less than the PC or PO . @ndushay @tingulfsen

Reply:
no - I don't think we want to update the version on either object (PO or PC) when incoming version < catalog version.

Reply:
To actually reply properly to Sarav's comment above:   no - we are only indicating that we have audited the pres copy;  nothing to change in the preservedObject in this happy path case.